File: python/raindrops/raindrops.py
def convert(number):
    """
    Convert a number into a string containing raindrop sounds corresponding to its factors.

    The function takes an integer 'number' and applies the following rules:
    - If 'number' has 3 as a factor, 'Pling' is added to the result.
    - If 'number' has 5 as a factor, 'Plang' is added to the result.
    - If 'number' has 7 as a factor, 'Plong' is added to the result.
    - If 'number' does not have 3, 5, or 7 as factors, the result is the digits of the number.

    Args:
        number (int): The input number to be converted.

    Returns:
        str: A string containing raindrop sounds or the digits of the input number.
    """
    result = ""
    
    if number % 3 == 0:
        result += "Pling"
    
    if number % 5 == 0:
        result += "Plang"
    
    if number % 7 == 0:
        result += "Plong"
    
    # If 'result' is still an empty string, there are no factors of 3, 5, or 7
    if not result:
        result = str(number)
    
    return result

File: javascript/raindrops/raindrops.js
export const convert = (number) => {
  // Initialize an empty string to store the raindrop sounds
  let sounds = '';

  // Check if the number has 3, 5 or 7 as a factor
  if (number % 3 === 0) {
    sounds += 'Pling';
  }
  if (number % 5 === 0) {
    sounds += 'Plang';
  }
  if (number % 7 === 0) {
    sounds += 'Plong';
  }

  // If the number does not have any of 3, 5 or 7 as a factor,
  // convert the number to a string and add it to the string
  if (sounds === '') {
    sounds = number.toString();
  }

  return sounds;
};
